src/ai/backend/manager/api/auth.py

```diff
@@ -499,6 +500,21 @@ async def _pipe_builder(r: Redis) -> RedisPipeline:
             },
             "is_admin": row["keypairs_is_admin"],
         }
+
+        def validate_ip(user: Mapping[str, Any]):
+            allowed_client_ip = user.get("allowed_client_ip", None)
+            if not allowed_client_ip or allowed_client_ip is None:
+                # raw_ip is None or [] - empty list
+                return
+            assert isinstance(allowed_client_ip, list)
+            raw_client_addr: str | None = request.headers.get("X-BackendAI-IP") or request.remote
```

Comment:
In current implementation, when user who has IP restriction enabled tries to access Backend.AI API through webserver, will always fail because request.remote value will always point to webserver's IP address. Thankfully webserver already provides original IP address under `X-Forwarded-For` header, we can benefit by using that header to determine original value. Also there seems to be no any other logic adding this header to request on client, so I think it's okay to remove parsing `X-BackendAI-IP` header.